public/js/validation.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+function setTime() {
+  var now = new Date();
+  var time = now.toLocaleTimeString();
+  $('#time').html('Welcome, it is now ' + time);
+}
+
+function validateLoginForm() {
+
+  var username = $('#username').val();
+  var password = $('#password').val();
+
+
+  if (username === '') {
+    $('#username').css('border', '1px solid red');
+    $('#username_error').text('username is required');
+    $('#username').focus();
+    return false;
+  } else {
+    $('#username_error').text('');
+  }
+
+  if (password === '') {
+    $('#password').css('border', '1px solid red');
+    $('#password_error').text('password is required');
+    $('#password').focus();
+    return false;
+  } else {
+    $('#password_error').text('');
+  }
+}
+
+function loginToAccount() {
+  // if valid true and call AJAX - GET /js/credentials.js
+  //   if credentials match, create object with isLoggedIn set to true
+  //   if credentials do not match, create object with isLoggedIn set to false and error set
+  //   to error from invalidLoginErrorMessage returned by AJAX GET
+  // return javascript object
+
+  var username = $('#username').val();
+  var password = $('#password').val();
+
+  if (username && password) {
```

Comment:
Instead of a big conditional here, it's a better practice to do a little one for the negative case, e.g.

```
if (!condition) {
  console.error("descriptive error message");
  return false;
}
```

This focuses the reader on what error you are looking for rather than having them have to think through why the positive condition exists with no negative condition.